packages/grafana-ui/src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { ReactNode, PureComponent } from 'react';
+import { ButtonSize } from '../Button/AbstractButton';
+import { Button } from '../Button/Button';
+import { GrafanaTheme } from '../../types/theme';
+import { ThemeContext } from '../../themes/ThemeContext';
+import { cx, css } from 'emotion';
+import tinycolor from 'tinycolor2';
+import { increaseContrast } from '../../utils/colors';
+import { selectThemeVariant } from '../../themes/selectThemeVariant';
+
+export interface ToggleButtonState {
+  untouched: boolean;
+  selected: boolean;
+  value?: any;
+  prevPropSelected: boolean;
+}
+
+export interface ToggleButtonProps {
+  onChange?: (value: any) => void;
+  selected: boolean;
+  value?: any;
+  className?: string;
+  children: ReactNode;
+  tooltip?: string;
+  key?: any;
+  size?: ButtonSize;
+  // To only base the buttons state on the props, rather than internal state
+  controlled?: boolean;
+}
+
+export class ToggleButton extends PureComponent<ToggleButtonProps, ToggleButtonState> {
+  constructor(props: ToggleButtonProps) {
+    super(props);
+    this.state = {
+      untouched: true,
+      selected: props.selected,
+      value: props.value,
+      prevPropSelected: props.selected,
+    };
+
+    this.handleBlur = this.handleBlur.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e: React.SyntheticEvent) {
+    e.stopPropagation();
+
+    this.setState(prevState => ({
+      untouched: false,
+      selected: this.props.controlled ? this.props.selected : !prevState.selected,
+    }));
+    if (this.props.onChange) {
+      this.props.onChange(this.state);
+    }
+  }
+
+  handleBlur() {
+    this.setState({
+      untouched: true,
+    });
+  }
+
+  static getDerivedStateFromProps(props: ToggleButtonProps, state: ToggleButtonState) {
+    if (props.selected !== state.prevPropSelected) {
+      return {
+        selected: props.selected,
+        prevPropSelected: props.selected,
+      };
+    }
+    return null;
+  }
+
+  renderButton(theme: GrafanaTheme) {
+    const styles = getStyles(theme);
+    // @ts-ignore
+    const { children, className, tooltip, controlled, ...otherProps } = this.props;
+    const activeStyle = this.state.selected ? styles.active : null;
+    return (
+      <Button
```

Comment:
wonder if it would be better to use radio inputs restyled instead, like in https://codepen.io/kaydelaney/pen/QWLyNRz

not sure what the benefit would be

Reply:
It feels cleaner and more semantically correct as Kay said. Should probably rename the component to reflect the radio functionality in that case, `RadioButton` and `RadioButtonGroup`

Reply:
Some benefits would be better semantics which is helpful for accessibility, as well as automatic keyboard control

Reply:
ok, lets do that, and rename it to RadioButton & RadioButtonGroup